server.go

```diff
@@ -1518,32 +1518,59 @@ func (s *server) genNodeAnnouncement(refresh bool,
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
+	// If we don't need to refresh the announcement, then we can return a
+	// copy of our cached version.
 	if !refresh {
 		return *s.currentNodeAnn, nil
 	}
 
+	// Now that we know we need to update our copy, we'll apply all the
+	// function updates that'll mutate the current version of our node
+	// announcement.
 	for _, update := range updates {
 		update(s.currentNodeAnn)
 	}
 
+	// We'll now update the timestamp, ensuring that with each update, the
+	// timestamp monotonically increases.
 	newStamp := uint32(time.Now().Unix())
 	if newStamp <= s.currentNodeAnn.Timestamp {
 		newStamp = s.currentNodeAnn.Timestamp + 1
 	}
-
 	s.currentNodeAnn.Timestamp = newStamp
+
+	// Now that the announce tn is fully updated, we'll generate a new
+	// signature over the announcement to ensure nodes on the network
+	// accepted the new authenticated announcement.
 	sig, err := discovery.SignAnnouncement(
 		s.nodeSigner, s.identityPriv.PubKey(), s.currentNodeAnn,
 	)
 	if err != nil {
 		return lnwire.NodeAnnouncement{}, err
 	}
-
 	s.currentNodeAnn.Signature, err = lnwire.NewSigFromSignature(sig)
 	if err != nil {
 		return lnwire.NodeAnnouncement{}, err
 	}
 
+	// Finally, we'll update the on-disk version of our announcement so it
```

Comment:
This commit should be reverted/squashed with the one that moves it to the tor controller.